unleash/Strategies/DefaultStrategy.swift

```diff
@@ -0,0 +1,18 @@
+//
+//  DefaultStrategy.swift
+//  Unleash
+//
+//  Copyright © 2019 Silvercar. All rights reserved.
+//
+
+import Foundation
+
+class DefaultStrategy : Strategy {
+    func getName() -> String {
+        return "default"
```

Comment:
Is this the best way to handle a magic string in Swift?

Reply:
I used to do something like a constants struct or an extension

Reply:
If I created an extension to String in this file, would it only be accessible for this file? I don't want a global constants file.

Reply:
I think private and fileprivate work with extensions if that's what you want to do

Reply:
@alamantia like this:
```
private extension String {
    static let name: String = "default"
}

class DefaultStrategy : Strategy {
    func getName() -> String {
        return .name
    }
    
    func isEnabled(parameters: [String : String]) -> Bool {
        return true
    }
}
```

Reply:
@deandalm Yes, like this. However, making a function “getName” isn’t really swift like when you can instead have a computed property that returns the string value.

Reply:
@louisvdaily here is the protocol I wrote, how should it be changed:
```
public protocol Strategy {
    func getName() -> String
    func isEnabled(parameters: [String: String]) -> Bool
}
```

Reply:
@deandalm I’m doing this from my phone, so please excuse me. You can do so like this:

var name: String { get }

In place of:

func getName() -> String

In use:

class DefaultStrategy: Strategy {
  let name: String {
    return “default”
  }

  // extra class stuff here
}